cmd/nitro/init.go

```diff
@@ -606,132 +606,108 @@ func rebuildLocalWasm(ctx context.Context, config *gethexec.Config, l2BlockChain
-				}
-				executionDB := rawdb.WrapDatabaseWithWasm(chainDB, wasmDB)
-				_, err = rawdb.ParseStateScheme(cacheConfig.StateScheme, executionDB)
-				if err != nil {
-					return nil, nil, err
-				}
-				err = pruning.PruneExecutionDB(ctx, executionDB, stack, &config.Init, cacheConfig, persistentConfig, l1Client, rollupAddrs, config.Node.ValidatorRequired(), false)
-				if err != nil {
-					return executionDB, nil, fmt.Errorf("error pruning: %w", err)
-				}
-				l2BlockChain, err := gethexec.GetBlockChain(executionDB, cacheConfig, chainConfig, tracer, &config.Execution.TxIndexer)
-				if err != nil {
-					return executionDB, nil, err
-				}
-				err = validateBlockChain(l2BlockChain, chainConfig)
-				if err != nil {
-					return executionDB, l2BlockChain, err
-				}
-				if config.Init.RecreateMissingStateFrom > 0 {
-					err = staterecovery.RecreateMissingStates(executionDB, l2BlockChain, cacheConfig, config.Init.RecreateMissingStateFrom)
-					if err != nil {
-						return executionDB, l2BlockChain, fmt.Errorf("failed to recreate missing states: %w", err)
-					}
-				}
-				return rebuildLocalWasm(ctx, &config.Execution, l2BlockChain, executionDB, wasmDB, config.Init.RebuildLocalWasm)
-			}
-			readOnlyDB.Close()
-		} else if !dbutil.IsNotExistError(err) {
-			// we only want to continue if the database does not exist
-			return nil, nil, fmt.Errorf("Failed to open database: %w", err)
+		// TODO: There will be a similar option for consensus-node=true: func openExistingConsensusDB(...).
+		executionDB, wasmDB, chainConfig, err := openExistingExecutionDB(stack, config, chainId, cacheConfig, persistentConfig)
+		if err != nil {
+			return executionDB, nil, err
 		}
-	}
+		// if is not able to read stored chain config then falls of the outer if, and continues to initialization by downloading snapshot
+		if executionDB != nil && wasmDB != nil {
+			// TODO: should this be included in openExistingExecutionDB?
+			err = pruning.PruneExecutionDB(ctx, executionDB, stack, &config.Init, cacheConfig, persistentConfig, l1Client, rollupAddrs, config.Node.ValidatorRequired(), false)
+			if err != nil {
+				return executionDB, nil, fmt.Errorf("error pruning: %w", err)
+			}
+			l2BlockChain, err := gethexec.GetBlockChain(executionDB, cacheConfig, chainConfig, tracer, &config.Execution.TxIndexer)
+			if err != nil {
+				return executionDB, nil, err
+			}
 
-	// Check if database was misplaced in parent dir
-	const errorFmt = "database was not found in %s, but it was found in %s (have you placed the database in the wrong directory?)"
-	parentDir := filepath.Dir(stack.InstanceDir())
-	if dirExists(path.Join(parentDir, "l2chaindata")) {
-		return nil, nil, fmt.Errorf(errorFmt, stack.InstanceDir(), parentDir)
-	}
-	grandParentDir := filepath.Dir(parentDir)
-	if dirExists(path.Join(grandParentDir, "l2chaindata")) {
-		return nil, nil, fmt.Errorf(errorFmt, stack.InstanceDir(), grandParentDir)
+			// TODO: should this be included in openExistingExecutionDB?
```

Comment:
Today nitro validates the blockchain for both cases, after downloading the DB, and after opening an existing DB.
So, to keep the current behavior, and not replicate code/execution, this should not be included in openExistingExecutionDB.